health.go

```diff
@@ -52,30 +51,77 @@ type ICheckable interface {
 	Status() (interface{}, error)
 }
 
-// The Config struct is used for defining and configuring checks.
+// IStatusListener is an interface that handles health check failures and
+// recoveries, primarily for stats recording purposes
+type IStatusListener interface {
+	// HealthCheckFailed is a function that handles the failure of a health
+	// check event. This function is called when a health check state
+	// transitions from passing to failing.
+	// 	* name - the name of the health check that failed
+	HealthCheckFailed(name string)
```

Comment:
Should this instead accept a `(state *State)` as an arg? Then the callback has all of the info in regards to the failure, rather than just name (and you could get fancier w/ what you do in the callback).

Reply:
Absolutely 👍